### ISSUE TITLE
Implement Element.scrollTop and Element.scrollLeft

### ISSUE THREAD

**Issue by peterjoel** [servo/servo#9650]
https://drafts.csswg.org/cssom-view/#dom-element-scrolltop
https://drafts.csswg.org/cssom-view/#dom-element-scrollleft

They are mentioned in [issue 1713](https://github.com/servo/servo/issues/1713#issue-27945054), but not actually included with the accepted PR.

**Comment by izgzhen** [comment]
I am interested to take this issue. But I am a new comer and wonder if it would be relatively easy to work out.

**Comment by peterjoel** [comment]
@izgzhen I looked into it a bit when I logged the issue. It can be done by only writing Rust (the bindings to JS are already in place). I suspect there won't be much code to write, most likely the difficulty will be in finding how to access the data you need.

**Comment by izgzhen** [comment]
@peterjoel Thanks. So should I simply copy & paste your stub code into my master branch servo or work on your forked branch? I favor the former one since it is simpler, but I am not sure about your idea.

**Comment by SimonSapin** [comment]
Note that these API can make elements with `overflow: hidden` scroll. Test case:

http://software.hixie.ch/utilities/js/live-dom-viewer/saved/3962
```html
<body onload='for (i=0; i<4; i++) document.getElementsByTagName("p")[i].scrollLeft = (i+1) * 20'>
<p style="overflow: hidden">Pseudopseudohypoparathyroidism
<p style="overflow: auto">Pseudopseudohypoparathyroidism
<p style="overflow: scroll">Pseudopseudohypoparathyroidism
<p style="overflow: visible">Pseudopseudohypoparathyroidism
<style>
p { border: solid; width: 100px

**Comment by peterjoel** [comment]
@izgzhen Create your own branch from master. Any related changes from my fork are already merged. (Unless I forgot something and you're talking about something specific?)

**Comment by izgzhen** [comment]
@peterjoel I mean this https://github.com/peterjoel/servo/commit/17b4253fe9fd59c8785469dda66c51d8145bd6bb change. It is not in master branch.

**Comment by peterjoel** [comment]
@izgzhen Ah, yes I forgot I did that. Yes, either copy it or pull from my fork into yours. Copying sounds easiest.

**Comment by timvandermeij** [comment]
Fixed by #9968?
